src/Data/Bin/Base.agda

```diff
@@ -0,0 +1,129 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Arithmetic of binary represented natural numbers. Initial part.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Bin.Base where
+
+import      Algebra.FunctionProperties as FuncProp
+open import Data.Nat using (ℕ) renaming (suc to 1+_; _*_ to _*ₙ_)
+open import Function        using (_∘_)
+open import Relation.Binary using (Decidable)
+open import Relation.Binary.PropositionalEquality using (_≡_; refl; cong)
+open import Relation.Nullary                      using (yes; no)
+
+
+data Bin : Set
+  where
+  0#    : Bin
+  2suc  : Bin → Bin    -- \n→ 2*(1+n)  arbitrary nonzero even
+  suc2* : Bin → Bin    -- \n→ 1 + 2n   arbitrary odd
+--
+-- This representation is unique for each natural number.
+
+
+open FuncProp (_≡_ {A = Bin}) using (Op₂)
+
+2suc-injective : {x y : Bin} → 2suc x ≡ 2suc y → x ≡ y
```

Comment:
These two lemmas + the decidability proof are properties of the equality relation over `Bin` and therefore, as in keeping with the rest of the standard library, belong in `Data.Bin.Properties`. By convention `Data.Bin` then re-exports the decidability proof. (see e.g. `Data.Nat`)

Reply:
Moved to Data.Bin.Properties.